### PR TITLE
chore: export operations moved into methods & added create index for gpkg & directory support

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**

### CODE REVIEW

File: confd/production.tmpl [hunk 1]
```diff
@@ -11,7 +11,8 @@
     "input_output": {
         "output_srs": "{{ getv "/input/output/output/srs" "EPSG:4326" }}",
         "output_format": "{{ getv "/input/output/output/format" "GPKG" }}",
-        "folder_path": "{{ getv "/input/output/folder/path" "./outputs" }}"
+        "folder_path": "{{ getv "/input/output/folder/path" "./outputs" }}",
+        "shared_folder": "{{ getv "/input/output/shared/folder" "default"}}"
```
Comment (asafMasa) [inline confd/production.tmpl]: shared_folder  ?

File: confd/production.toml [hunk 2]
```diff
@@ -12,6 +12,7 @@ keys = [
     "/input_output/output_srs",
     "/input_output/output_format",
     "/input_output/folder_path",
+    "/input_output/shared_folder",
```
Comment (asafMasa) [inline confd/production.toml:14]: shared_folder ?

File: src/helper.py [hunk 3]
```diff
@@ -26,19 +27,47 @@ def json_fields_validate(self, json_obj):
...
+    def valid_configuration(self, keys):
+        value = self.__config[keys[0]][keys[1]]
+        if value:
+            self.logger.info(f'{keys[1]} is set to {value}')
+        else:
+            raise ValueError(f'Bad Configuration - no value for {keys[1]} variable.')
+
+    def create_folder_if_not_exists(self, path):
+        try:
+            mkdir(path)
```
Comment (asafMasa) [inline src/helper.py]: check if folder exists

File: src/exportImage.py [hunk 4]
```diff
@@ -13,52 +13,47 @@ def __init__(self):
...
             if result is not None:
-                link = f'{self.__config["input_output"]["folder_path"]}/{filename}.gpkg'
-                doc = {
-                    "taskId": taskid,
-                    "status": Status.COMPLETED.value,
-                    "progress": 100,
-                    "lastUpdateTime": datetime.utcnow(),
-                    "link": link
-                }
+                #TODO: when trigger update directory name, change link.
```
Comment (asafMasa) [inline src/exportImage.py]: remove

File: src/exportImage.py [hunk 5]
```diff
@@ -13,52 +13,47 @@ def __init__(self):
...
+        self.logger.info(f'Task Id "{taskid}" in progress.')
+        kwargs = {'dstSRS': self.__config['input_output']['output_srs'],
+                  'format': self.__config['input_output']['output_format'],
+                  'outputBounds': bbox,
+                  'callback': self.progress_callback,
+                  'callback_data': es_obj,
+                  'xRes': 1.67638063430786e-07,
+                  'yRes': 1.67638063430786e-07,
+                  'creationOptions': ['TILING_SCHEME=InspireCrs84Quad']}
+        result = gdal.Warp(f'{self.__config["input_output"]["folder_path"]}/{directoryName}/{filename}.gpkg', url, **kwargs)
```
Comment (asafMasa) [inline src/exportImage.py]: check if you have path.join

File: src/exportImage.py [hunk 6]
```diff
@@ -13,52 +13,47 @@ def __init__(self):
...
-                link = f'{self.__config["input_output"]["folder_path"]}/{filename}.gpkg'
-                doc = {
-                    "taskId": taskid,
-                    "status": Status.COMPLETED.value,
-                    "progress": 100,
-                    "lastUpdateTime": datetime.utcnow(),
-                    "link": link
-                }
+                #TODO: when trigger update directory name, change link.
+                link = f'{self.__config["input_output"]["folder_path"]}/{directoryName}/{filename}.gpkg'
```
Comment (asafMasa) [inline src/exportImage.py]: link -> rename to full path

File: src/exportImage.py [hunk 7]
```diff
@@ -13,52 +13,47 @@ def __init__(self):
...
 
-                self.__helper.update_db(doc)
+                self.create_index(filename, link)
```
Comment (asafMasa) [inline src/exportImage.py]: open issue to update the uml flow

File: src/helper.py [hunk 8]
```diff
@@ -26,19 +27,47 @@ def json_fields_validate(self, json_obj):
...
+        if progress is not None:
+            doc["progress"] = progress
+        if link is not None:
+            doc["link"] = link
+
         try:
             headers = {"Content-Type": "application/json"}
 
-            self.logger.info(f'Task Id "{doc["taskId"]}" Updating database')
+            self.logger.info(f'Task Id "{taskId}" Updating database')
```
Comment (asafMasa) [inline src/helper.py]: add doc to log

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: confd/production.tmpl (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply confd/production.tmpl]: fixed on 70a2a43

File: confd/production.toml (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply confd/production.toml:14]: fixed on 70a2a43

File: src/helper.py (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply src/helper.py]: fixed on f400a9f

File: src/exportImage.py (hunk 4 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: removed on 29dbb84

File: src/exportImage.py (hunk 5 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: fixed on ce6e4c4

File: src/exportImage.py (hunk 6 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: fixed on a897dbd

File: src/exportImage.py (hunk 7 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: issue created

File: src/helper.py (hunk 8 above)
Comment (CL-SHLOMIKONCHA) [reply src/helper.py]: fixed on caa7e25

**Review by asafMasa — APPROVED** [review]

**Pull request merged**